TransbankPosSDK/POSAutoservicio.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using Transbank.Utils;
+using Transbank.Exceptions.CommonExceptions;
+using System.Text;
+using Transbank.Responses.CommonResponses;
+using Transbank.Responses.AutoservicioResponse;
+using Transbank.Exceptions.AutoservicioExceptions;
+using System.Threading.Tasks;
+
+namespace Transbank.POSAutoservicio
+{
+    public class POSAutoservicio : Serial
+    {
+        public POSAutoservicio()
+        {
+
+        }
+
+        public static POSAutoservicio Instance { get; } = new POSAutoservicio();
+
+        public async Task<bool> Poll()
+        {
+            DiscardBuffer();
+
+            if (CantWrite())
+            {
+                throw new TransbankException($"Unable to Poll port {Port.PortName} is closed");
+            }
+
+            try
+            {
+                byte[] buffer = new byte[1];
+                string command = "0100";
+
+                Console.WriteLine($"Out (Hex): {ToHexString(command)}");
+                Console.WriteLine($"Out (ASCII): {command}");
+
+                Port.Write("0100");
+                await Port.BaseStream.ReadAsync(buffer, 0, 1);
+                return CheckACK(buffer[0]);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankException($"Unable to send Poll command on port {Port.PortName}", e);
+            }
+        }
+
+        public async Task<LoadKeysResponse> LoadKeys()
+        {
+            try
+            {
+                await WriteData("0800");
+                return new LoadKeysResponse(CurrentResponse);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankLoadKeysException("Unable to execute Load Keys in pos", e);
+            }
```

Comment:
I think that if you do this, you are loosing the original exception message, so it would be more difficult to know why it failed?

Reply:
All the exceptions of the port writing are included and they are added as a detail of the method exception